apps/studio/components/interfaces/DataWarehouse/WarehouseAccessTokens.tsx

```diff
@@ -0,0 +1,239 @@
+  Input,
+  cn,
+} from 'ui'
+import { GenericSkeletonLoader } from 'ui-patterns'
+import CreateWarehouseAccessToken from './CreateWarehouseAccessToken'
+
+const AccessTokenItem = ({
+  token,
+  id,
+  description,
+  onDeleteClick,
+  inserted_at,
+}: {
+  token: string
+  id: number
+  description: string
+  inserted_at: string
+  onDeleteClick: (id: number) => void
+}) => {
+  const formattedInsertedAt = new Date(inserted_at).toLocaleString()
+
+  return (
+    <Table.tr className="group">
+      <Table.td className="max-w-[240px]">{description}</Table.td>
+      <Table.td>{formattedInsertedAt}</Table.td>
+      <Table.td>
+        <div className="flex gap-1 relative">
+          <Input disabled defaultValue={token} type="password" size="tiny" className="flex-grow" />
+          <div className="w-[88px]">
+            <CopyButton
+              type="outline"
+              text={token}
+              className="group-hover:opacity-100 opacity-0 transition-opacity absolute right-0"
+            />
+          </div>
+        </div>
+      </Table.td>
+      <Table.td className="!p-1.5 text-right">
+        <DropdownMenu>
+          <DropdownMenuTrigger className="h-8 w-8 p-2 focus-visible:outline-none">
+            <MoreVertical size="14" />
+          </DropdownMenuTrigger>
+          <DropdownMenuContent className="max-w-40">
+            <DropdownMenuItem
+              className="flex gap-1.5 "
+              onClick={(e) => {
+                e.preventDefault()
+                onDeleteClick(id)
+              }}
+            >
+              <TrashIcon size="14" />
+              Revoke token
+            </DropdownMenuItem>
+          </DropdownMenuContent>
+        </DropdownMenu>
+      </Table.td>
+    </Table.tr>
+  )
+}
+
+const WarehouseAccessTokens = () => {
+  const isProjectActive = useIsProjectActive()
+  const params = useParams()
+  const projectRef = params.ref as string
+  const [open, setOpen] = useState(false)
+
+  const accessTokensQuery = useWarehouseAccessTokensQuery({ projectRef })
+  const hasAccessTokens = accessTokensQuery.isSuccess && accessTokensQuery.data.data.length > 0
+  const [tokenToDelete, setTokenToDelete] = useState<string | null>(null)
+  const [showDeleteDialog, setShowDeleteDialog] = useState(false)
+
+  const { mutate: deleteWarehouseAccessToken, isLoading: deleteLoading } =
+    useDeleteWarehouseAccessToken({
+      onSuccess: () => {
+        toast.success('Access token revoked')
+      },
+      onError: () => {
+        toast.error('Failed to revoke access token')
+      },
+    })
+
+  const { mutate: createWarehouseAccessToken, isLoading: createLoading } =
+    useCreateWarehouseAccessToken({
+      onSuccess: () => {
+        toast.success('Access token created')
+        setOpen(false)
+      },
+    })
+
+  const isLoading = deleteLoading || createLoading
+
+  return (
+    <div className="1xl:px-28 mx-auto flex flex-col gap-y-10 px-5 py-6 lg:px-16 xl:px-24 2xl:px-32 pb-32">
```

Comment:
seems like this div might be unnecessary - can pass all the classNames to section instead